From a03bf02eb3d9df6fb5125ddc295c6c5097a24195 Mon Sep 17 00:00:00 2001
From: DennySPB <dennyspb@gmail.com>
Date: Tue, 18 Sep 2018 13:04:00 +0300
Subject: [PATCH 1/6] Move QC board variables earlie

Squash of:

lineage: Move QC board variables earlier

These variables are usually set in a device's BoardConfig, setting them
in qcom_target is at the very end of the "configuration process" which
results in them being unavailable to plenty of other configuration
"things" (ex. soong namespaces or soong config). Move them to right
after a device's BoardConfig has been found and loaded.

Change-Id: Iddd731202d22ed3f8eb010197ce20d3c75a1f40a

qcom: Declare PRODUCT_SOONG_NAMESPACES for HALs

HALs are starting to move to blueprint files, which means that pathmap
ifeqs no longer work. This results in modules with the same name in
different platform HAL directories. Since these modules aren't used or
linked anywhere outside the HAL directories and a device's device.mk,
we can safely use a soong namespace to isolate the different HAL
directories' blueprints. This defines the PRODUCT_SOONG_NAMESPACES
make flag so make knows which namespace to look in as well.

Change-Id: I549e6da863465d79b12c7453aa169d5d963afa52
---
 build/core/BoardConfigQcom.mk | 65 +++++++++++++++++++++++++++++++++++
 1 file changed, 65 insertions(+)
 create mode 100644 build/core/BoardConfigQcom.mk

diff --git a/build/core/BoardConfigQcom.mk b/build/core/BoardConfigQcom.mk
new file mode 100644
index 00000000..0667a774
--- /dev/null
+++ b/build/core/BoardConfigQcom.mk
@@ -0,0 +1,65 @@
+# Bring in Qualcomm helper macros
+include vendor/havoc/build/core/qcom_utils.mk
+
+B_FAMILY := msm8226 msm8610 msm8974
+B64_FAMILY := msm8992 msm8994
+BR_FAMILY := msm8909 msm8916
+UM_3_18_FAMILY := msm8937 msm8953 msm8996
+UM_4_4_FAMILY := msm8998 sdm660
+
+BOARD_USES_ADRENO := true
+
+#TARGET_USES_QCOM_BSP := true
+
+# Tell HALs that we're compiling an AOSP build with an in-line kernel
+TARGET_COMPILE_WITH_MSM_KERNEL := true
+
+ifneq ($(filter msm7x27a msm7x30 msm8660 msm8960,$(TARGET_BOARD_PLATFORM)),)
+    # Enable legacy audio functions
+    ifeq ($(BOARD_USES_LEGACY_ALSA_AUDIO),true)
+        USE_CUSTOM_AUDIO_POLICY := 1
+    endif
+endif
+
+# Allow building audio encoders
+TARGET_USES_QCOM_MM_AUDIO := true
+
+# Enable color metadata for UM platforms
+ifeq ($(call is-board-platform-in-list, $(UM_3_18_FAMILY) $(UM_4_4_FAMILY)),true)
+    TARGET_USES_COLOR_METADATA := true
+endif
+
+# List of targets that use master side content protection
+MASTER_SIDE_CP_TARGET_LIST := msm8996 msm8998 sdm660
+ifeq ($(call is-board-platform-in-list, $(B_FAMILY)),true)
+    MSM_VIDC_TARGET_LIST := $(B_FAMILY)
+    QCOM_HARDWARE_VARIANT := msm8974
+else
+ifeq ($(call is-board-platform-in-list, $(B64_FAMILY)),true)
+    MSM_VIDC_TARGET_LIST := $(B64_FAMILY)
+    QCOM_HARDWARE_VARIANT := msm8994
+else
+ifeq ($(call is-board-platform-in-list, $(BR_FAMILY)),true)
+    MSM_VIDC_TARGET_LIST := $(BR_FAMILY)
+    QCOM_HARDWARE_VARIANT := msm8916
+else
+ifeq ($(call is-board-platform-in-list, $(UM_3_18_FAMILY)),true)
+    MSM_VIDC_TARGET_LIST := $(UM_3_18_FAMILY)
+    QCOM_HARDWARE_VARIANT := msm8996
+else
+ifeq ($(call is-board-platform-in-list, $(UM_4_4_FAMILY)),true)
+    MSM_VIDC_TARGET_LIST := $(UM_4_4_FAMILY)
+    QCOM_HARDWARE_VARIANT := msm8998
+else
+    MSM_VIDC_TARGET_LIST := $(TARGET_BOARD_PLATFORM)
+    QCOM_HARDWARE_VARIANT := $(TARGET_BOARD_PLATFORM)
+endif
+endif
+endif
+endif
+endif
+
+PRODUCT_SOONG_NAMESPACES += \
+  hardware/qcom/audio-caf/$(QCOM_HARDWARE_VARIANT) \
+  hardware/qcom/display-caf/$(QCOM_HARDWARE_VARIANT) \
+  hardware/qcom/media-caf/$(QCOM_HARDWARE_VARIANT)
\ No newline at end of file
-- 
2.19.1

